homeassistant/components/bluetooth/entity.py

```diff
@@ -0,0 +1,99 @@
+"""The bluetooth integration."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+import dataclasses
+from typing import TYPE_CHECKING
+
+from homeassistant.components import bluetooth
+from homeassistant.core import callback
+from homeassistant.helpers import device_registry as dr, entity
+
+if TYPE_CHECKING:
+    from .update_coordinator import BluetoothDataUpdateCoordinator
+
+
+@dataclasses.dataclass(frozen=True)
+class BluetoothDeviceKey:
+    """Key for a bluetooth device.
+
+    Example:
+    device_id: outdoor_sensor_1
+    key: temperature
+    """
+
+    device_id: str | None
+    key: str
+
+
+BluetoothDeviceEntityDescriptionsType = Mapping[
```

Comment:
Why do we need this shortcut? I was confused the first time I read the coordinator code as the type didn't match what I was expecting (a mapping).

Reply:
d080a6c4cbaeafc11fad2fb17423173a46509e72